apps/api/src/app/workflows-v2/shared/parse-payload-schema.ts

```diff
@@ -0,0 +1,19 @@
+import { JSONSchemaDto } from '@novu/shared';
+
+export function parsePayloadSchema(schema: unknown): JSONSchemaDto {
+  if (typeof schema === 'string') {
+    try {
+      return JSON.parse(schema);
+    } catch (error) {
+      throw new Error(
+        'Invalid payload schema: Unable to parse provided JSON string. Please ensure the string contains valid JSON syntax.'
+      );
+    }
+  }
+
+  if (schema && typeof schema === 'object') {
```

Comment:
Let's check this first and return the object. 
```suggestion
  if (typeof schema === 'object') {
```

Then we can try/catch with `JSON.parse` on anything else without extra ifs. Please also make sure to add a type assertion on JSON.parse.

Reply:
I think if it don't exist it will say that typeof is undefined so it's not needed